### PR TITLE
Fix problem with x-range processing

### PR DESCRIPTION
**Pull request by schreiberste**
Fixes issue https://github.com/semver4j/semver4j/issues/137

### CODE REVIEW

File: src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java [hunk 1]
```diff
@@ -64,9 +61,12 @@ public String process(String range) {
                         compareSign = LT.asString();
                         if (isX(minor)) {
                             major = major + 1;
+                            minor = 0;
```
Comment (piotrooo) [inline src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java:64]: We definitively need test case for this. You could add new test in https://github.com/semver4j/semver4j/blob/e2c265f9f417b0e3d159da9113e695fe4d980752/src/test/java/org/semver4j/SemverTest.java#L851 as an additional arguments around `Comparators` section.

**Review by piotrooo — CHANGES_REQUESTED** [review]

File: src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java (hunk 1 above)
Comment (schreiberste) [reply src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java:64]: Tests added.

**Comment by piotrooo** [comment]
Many thanks with this nasty bug...

**Comment by schreiberste** [comment]
Thanks for merging and the fast new release!

**Pull request merged**